### PR TITLE
Merged _bool from axis into cbook._string_to_bool

### PR DESCRIPTION
**Pull request by Kojoley**

### CODE REVIEW

File: lib/matplotlib/cbook.py [hunk 1]
```diff
@@ -784,14 +784,15 @@ def is_scalar_or_string(val):
     return is_string_like(val) or not iterable(val)
 
 
-def _string_to_bool(s):
+def string_to_bool(s):
+    """Parses the string argument as a boolean"""
```
Comment (efiring) [inline lib/matplotlib/cbook.py]: For this to be a general cbook function, I think it should accept case-insensitive "true" and "false" as well as "on" and "off".

Comment (Kojoley) [inline lib/matplotlib/cbook.py]: Yes, I thought about it. I will add them then.

**Comment by tacaswell** [comment]
This makes the kwargs a bit more permissive in what they accept (which seems ok).

Why make the function public?

**Comment by Kojoley** [comment]
Any reasons why it should not? I think that private functions should not be used outside of the module while this function is used in multiple external places.

**Comment by Kojoley** [comment]
I see no problem with making it public as it is quite simple and should not be changed frequently (or even at all). If my position is wrong then I will remake it private.

**Comment by WeatherGod** [comment]
You'd be surprise what changes. More precisely, you'd be surprised how many
people complain when old, unused functions are removed.  Since this
function does not have anything to do with plotting, it would make sense to
keep it private to keep people from depending on matplotlib for
non-plotting use-cases.

**Comment by Kojoley** [comment]
Hmm. I see your point and it is strong, but the whole `cbook` is not about plotting and most functions it contains are public... why module itself is public then?

**Comment by WeatherGod** [comment]
We were young... we were foolish...

**Comment by tacaswell** [comment]
Also, I suspect the existence of `cbook` pre-dates the convention that a leading underscore means 'private'.

**Comment by dopplershift** [comment]
Well, I think `cbook` also started as a landing site for things borrowed from places like: https://code.activestate.com/recipes/langs/python/

**Comment by tacaswell** [comment]
@Kojoley Thanks!

**Pull request merged**